Fix Windows builds

hwsmm opened the pull request:
- Refer to https://github.com/ppy/SDL3-CS/pull/193#issuecomment-2567045896

Not sure if it works because the workflow refuses to run on a forked repository..? I also added a bash option to exit on a command error.

Susko3 commented:
You can run the workflow like this on your repo. Update your `master` to upstream (`ppy/SDL3-CS`) if it doesn't appear.

![slika](https://github.com/user-attachments/assets/d0b0574a-f98d-47e3-8e83-c1039017784e)

hwsmm commented on External/build.sh:
```diff
@@ -88,6 +90,12 @@ fi
+
+if [[ $RUNNER_OS == 'Windows' ]]; then
+    echo "Patching SDL to not include gameinput.h"
+    patch -Np1 -i ../exclude-gameinput.patch
```
I'll apply if you find out how to do it without a patch, though I'm not sure why we need to avoid a patch even for a workaround.

hwsmm commented on External/build.sh:
```diff
@@ -8,7 +10,7 @@ if [[ -z $NAME || -z $RUNNER_OS || -z $FLAGS ]]; then
 fi
 
-SUDO=$(which sudo)
+SUDO=$(which sudo || exit 0)
```
`which sudo` fails on Windows and returns non-zero exit code. `set -e` makes bash quit on such exit code, so this is needed.

hwsmm commented:
I can run the workaround, but it gives me this error: `actions/checkout@v4, actions/upload-artifact@v4, actions/setup-java@v4, android-actions/setup-android@v3, actions/download-artifact@v4, and 4 other actions are not allowed to be used in hwsmm/SDL3-CS. Actions in this workflow must be: within a repository owned by hwsmm.`

I guess this is because I'm just a collaborator here?

smoogipoo commented:
I don't think that's something I can change. You probably have to go into Settings > Actions > General on your fork and check the permissions there.

Susko3 reviewed (dismissed):
Sure, this can go in so we can get a build out. But if the patch fails to apply in the next SDL update, I'll be ripping it out and looking for an alternative solution.

Susko3 commented:
[arm build is failing](https://github.com/Susko3/SDL3-CS/actions/runs/12575217257/job/35049979407), has it ever worked?

hwsmm commented:
> You probably have to go into Settings > Actions > General on your fork and check the permissions there.

I did it before in SDL2-CS, but I forgot to do this in SDL3-CS 😅, thanks for the solution!

linux-arm seems to be failing due to file permission being mixed up with QEMU. We can probably set permission in build.sh, but we will get mixed permission again after building in this way, so I just added a commit to ignore the error on `git reset`.

I also replaced the patch with sed, and fixed formatting in the script.
